### DISCUSSION TITLE
課題14についての質問（8/17)

### DISCUSSION THREAD

**Discussion by TakedaAyano** [nsc-kiwamu/traning-takeda#71] [Q&A]
～問題１～
let ans = judgeSosuu()の()に入るものは、(i)だと思うのですがどうでしょうか？
なぜなら、console.log(num)または(100)として出力すると、ただ「100」が98回出力されただけでしたので(num),(100)ではないと思いました。
次にconsole.log(i)とすると2～99まで出力されたため、let ans = judgeSosuu(i)であると考えました。
しかし、なぜ100まで出力されないのかはわかりません。

～関数部分～
「target」が、素数判定したい値ですのでfor文・if文の中に入れました。
この関数部分のコードは合っていますでしょうか？

function judgeSosuu(target: number):number | undefined {
  let retNum:number | undefined

  // 素数判定の処理を追加する
  for(let i=2; i < target; i++){
        if(target%i == 0){ // target(受け取った値）÷iの余りがなければ、
              break; // 終了
        }
    if(target == i) { // target(受け取った値）とiが同じであれば、素数
    }
  }
  return retNum
}

**Comment by nsc-kiwamu** [comment]
・なぜ100まで出力されないのか
おそらく配列のループと混同しているのではないかと思います。
例えば、要素100の配列のループであれば、
for (let i = 0; i < list.length; i++)
ですよね？
これは配列の要素が0から始まるためlist[0]～list[99]
までアクセスしたいので、このような条件となります。
配列の終了条件は「;」で区切られた条件の真ん中のところになるので
「なぜ100まで出力されないのか」はどうすればよいかわかりますよね？

・関数部分の処理
考え方は合っています。
この関数の仕様として
素数の場合：数値を返す
素数以外の場合：undefindを返す
と決めたかと思います。

この処理ができているか見直してください
